custom-vu/src/main/kotlin/jces1209/vu/page/CloudAddScreenShot.kt

```diff
@@ -29,4 +29,11 @@ class CloudAddScreenShot(
             "data-test-status",
             "complete")
     }
+
+    override fun openScreenShot() {
+        getFirstScreenShot()?.click()
```

Comment:
don't like this null check - so we will have an exception
let's make it with list
```
protected fun getScreenShotsCount(): Int {
    return getScreenShots().size
}

protected fun getScreenShots(): List<WebElement>{
    return driver.findElements(screenShotLocator)
}
```
```suggestion
        val screenShots = getScreenShots()
        if (screenShots.isNotEmpty()) {
            screenShots
                .first()
                .click()
                
                ..... and other waits
        } else {
             logger.debug("There are no screenshots to open")
        }
```

Reply:
Note that `?.` will not throw exceptions, it's a [safe-call](https://kotlinlang.org/docs/reference/null-safety.html#safe-calls). The worst it can do is return `null`.